coffea/processor/executor.py

```diff
@@ -1087,6 +1133,13 @@ def run_uproot_job(
 
     if pre_executor is None:
         pre_executor = executor
+
```

Comment:
Here we could do:
```python
if isinstance(executor, type) and issubclass(executor, ExecutorABC):
    executor = executor()  # initialize with default arguments
elif not isinstance(executor, ExecutorABC):
    raise ValueError("...")
```
I think then we would eventually want to deprecate the first option and mandate users add `()` at the end of their executors to initialize them. This could come along with dropping `executor_args`.